File: libs/API/model/search_request.py
from libs.API.model.search_filter import SearchFilter


class SearchRequest(dict):
    page = 0
    pageSize = 50

    def __init__(self, **kwargs):
        super(SearchRequest, self).__init__(pageSize=self.pageSize,
                                            page=self.page, **kwargs)

    @property
    def filters(self):
        return [filtr for filtr in self.values() if (isinstance(filtr, SearchFilter) and filtr.value)]

File: api_tests/src/main/java/http/requests/SearchRequest.java
package http.requests;

import http.HttpMethod;
import model.CBSearchFilter;

public class SearchRequest extends HttpRequest {

    private final static String URI = "/api/search/search/";

    /**
     * Build HTTP request.
     * By Default: HttpMethod = GET, MediaType = APPLICATION_JSON.
     */
    public SearchRequest() {
        super(URI);
    }

    /**
     * CB Search
     * API: POST /api/search/search
     * payload: {"keywords":"67576","pageSize":20,"page":0}
     */
    public SearchRequest search(CBSearchFilter filter) {
        this
                .setURI(URI)
                .setHttpMethod(HttpMethod.POST)
                .setPayload(filter);
        return this;
    }
}
